### DISCUSSION TITLE
Why are tags such as net.peer.addr/net.sock.peer.addr/net.peer.name not in the span of kafka and rocketmq?

### DISCUSSION THREAD

**Discussion by tuhao1020** [open-telemetry/opentelemetry-java-instrumentation#10133] [Q&A]
net.sock.peer.addr and net.sock.peer.port are in the span of rabbitmq
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26807520/c1b970ab-9c92-48de-9a61-7644924b191a)

but, no such tags for kafka and rocketmq
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26807520/a2e05532-f5fa-4d7c-922e-9678b87698fc)
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26807520/96244e91-387e-4dd3-a226-761033a2c678)

is it a bug?

**Comment by laurit** [comment]
> is it a bug?

As https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/messaging.md#messaging-attributes lists `net.sock.peer.addr` and `net.sock.peer.port` as recommended not required it is not a bug.

**Comment by tuhao1020** [comment]
I want add `net.sock.peer.addr` and `net.sock.peer.port` tag when using kafka,  is there any way to do this? @laurit

**Reply by laurit** [reply]
It is hard to say what is needed for this without actually attempting to implement it. My guess is that if adding these were easy they would have been already added, I'd expect this to be tricky.

**Comment by tuhao1020** [comment]
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/26807520/a9792a33-1ae7-476a-b491-01353f8fd31e)

@laurit I find that rabbitmq `net.sock.peer.addr` and `net.sock.peer.port` exist in producer span, but not exist in consumer span 😭
